aiida/tools/archive/imports.py

```diff
@@ -195,35 +208,45 @@ def import_archive(
 
 def _add_new_entities(
     etype: EntityTypes, total: int, unique_field: str, backend_unique_id: dict, backend_from: StorageBackend,
-    backend_to: StorageBackend, batch_size: int, transform: Callable[[dict], dict]
+    backend_to: StorageBackend, qparams: QueryParams, transform: Callable[[dict], dict]
 ) -> None:
     """Add new entities to the output backend and update the mapping of unique field -> id."""
     IMPORT_LOGGER.report(f'Adding {total} new {etype.value}(s)')
-    iterator = QueryBuilder(backend=backend_from).append(
-        entity_type_to_orm[etype],
-        filters={
-            unique_field: {
-                '!in': list(backend_unique_id)
-            }
-        } if backend_unique_id else {},
-        project=['**'],
-        tag='entity'
-    ).iterdict(batch_size=batch_size)
+
+    # collect the unique entities from the input backend to be added to the output backend
+    ufields = []
```

Comment:
Do I understand correctly that now here you read in all the unique fields for this entity into memory from the input backend, whereas before this was directly streamed into the `bulk_insert`. Could this lead to memory problems, given that this very fix is to deal with large imports?